wikimedia/wikimedia-page-library
# Add/pcs abstraction layer

For mobile-html users let's hide some of the complexity of page library.
Instead we want to have an abstraction layer that can be backwards
and forwards compatible, and also simplifies usage.

The first functions exposed are in the PageMods module. Documentation
for the new functionality can be found in the new file docs/pcs/pcs.md.

The part that is missing from the associated task is
textSizeAdjustmentPercentage, which isn't implemented as of yet.
(See https://phabricator.wikimedia.org/T222752)

To test:

1) follow the instructions in the readme
(npm link && npm run -s build:debug; cd ../mobileapps;
npm link wikimedia-page-library; npm start

2) Open a local mobile-html page,
  e.g. http://localhost:6927/en.wikipedia.org/v1/page/mobile-html/Dog
  Note: make sure the old version is not cached (e.g. Shift-Refresh)

3) Open DevConsole, then try some of the calls mentioned in
docs/pcs/pcs.md

Bug: https://phabricator.wikimedia.org/T218049
Bug: https://phabricator.wikimedia.org/T206502
